### PR TITLE
Use nox as the top-level test runner, use gcprepotools for req sessions

### PR DESCRIPTION
**Pull request by jonparrott**
Note: has #32 as a diffbase (until it gets merged).

### CODE REVIEW

File: optional-container-engine/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```
Comment (waprin) [inline optional-container-engine/tests/conftest.py]: I'm confused why this is part of this PR.

File: 7-gce/tox.ini [hunk 2]
```diff
@@ -8,17 +8,23 @@ deps =
   -rrequirements.txt
   -rrequirements-dev.txt
 commands =
-  nosetests \
-    --with-flaky \
-    --no-success-flaky-report \
-    --with-coverage \
-    --cover-package bookshelf \
-    {posargs:-a '!e2e'}
+  py.test --cov=bookshelf --no-success-flaky-report -m "not e2e" {posargs: tests}
```
Comment (waprin) [inline 7-gce/tox.ini]: do we still need all the tox stuff?

File: optional-container-engine/tests/conftest.py (hunk 1 above)
Comment (jonparrott) [inline optional-container-engine/tests/conftest.py]: It had a diffbase, hold on, let me update now that the other PR is merged.

File: 7-gce/tox.ini (hunk 2 above)
Comment (jonparrott) [inline 7-gce/tox.ini]: I'm not going to rip tox out of the individual steps. I still think that most python developers will expect to see tox. We'll use nox to orchestrate running all of them together though.

File: .gitignore [hunk 3]
```diff
@@ -14,6 +14,7 @@ pip-delete-this-directory.txt
 # Unit test / coverage reports
 htmlcov/
 .tox/
+.nox/
```
Comment (waprin) [inline .gitignore]: super nit - can get rid of .tox here

**Comment by waprin** [comment]
lgtm

File: .gitignore (hunk 3 above)
Comment (jonparrott) [inline .gitignore]: Unfortunately we can't, because the individual steps still use tox and will create `.tox` folders. ;)

**Pull request merged**